pgmpy/metrics/metrics.py

```diff
@@ -467,3 +467,242 @@ def SHD(true_model, est_model):
     shd = shd + (np.sum((d + d.T) > 0) / 2)
 
     return int(shd)
+
+
+def _latent_admg(dag: DAG, observed: list) -> nx.DiGraph:
+    """
+    Compute the latent‐projection ADMG L(G, observed_set) of a DAG G onto a subset observed_set ⊂ V.
+    (Helper function for self_compatibility_graphical)
+
+    Implements Definition 5 (latent ADMG) from [1] (Faller et al., AISTATS 2024):
+    Let G be an ADMG with variables V and observed_set ⊂ V . The latent ADMG L(G, observed_set)
+    is the ADMG that contains all nodes in observed_set,
+    all edges between nodes in observed_set and additionally
+    1) a directed_edges edge between X, Y ∈ observed_set if there is a directed_edges path from X to Y where all intermediate
+        nodes are in V \ observed_set
+    2) a bidirected_edges edge between X, Y if there is a (undirected) path such that every non-endpoint is a noncollider in V \ observed_set and there are arrowheads towards
+        X and Y on the incident edges on the path.
+
+    Parameters
+    ----------
+    dag : DAG
+        The full DAG G on variables V (may include latent nodes).
+    observed : list
+        Subset observed_set ⊂ V to project onto (observed variables).
+
+    Returns
+    -------
+    nx.DiGraph
+    An ADMG over the observed nodes, where:
+        - X→Y encodes a latent‐only directed chain X→…→Y
+        - X↔Y is encoded by having both X→Y and Y→X in the grap.
+
+    >>> from pgmpy.models import DiscreteBayesianNetwork
+    >>> from pgmpy.metrics.metrics import _latent_admg
+    >>> # Example 1: A → H → B, observe only A,B
+    >>> dag1 = DiscreteBayesianNetwork([('A', 'H'), ('H', 'B')])
+    >>> admg1 = _latent_admg(dag1, observed=['A', 'B'])
+    >>> set(admg1.edges()) == {('A', 'B')}
+    True
+
+    >>> # Example 2: A → H ← B, observe only A,B
+    >>> dag2 = DiscreteBayesianNetwork([('A', 'H'), ('B', 'H')])
+    >>> admg2 = _latent_admg(dag2, observed=['A', 'B'])
+    >>> set(admg2.edges()) == {('A', 'B'), ('B', 'A')}
+    True
+
+    References
+    ----------
+    [1] Faller, P. M., et al. (2024).
+    “Self‐compatibility: Evaluating Causal Discovery without Ground Truth.”
+    In AISTATS. arXiv:2307.09552
+    """
+    full_directed = dag
+    variables = set(dag.nodes())
+    observed_set = set(observed)
+    latent_set = variables - observed_set
+
+    directed_edges = set()
+    bidirected_edges = set()
+
+    # 1) Preserve original observed→observed
+    for u, v in dag.edges():
+        if u in observed_set and v in observed_set:
+            directed_edges.add((u, v))
+
+    # 2) Rule 1: latent-only directed chains via active_trail_nodes
+    for u, v in combinations(observed_set, 2):
+        # Condition on all other observed nodes so only paths via latents remain
+        conditioning_set = list(observed_set - {u, v})
```

Comment:
Conditioning wouldn't do this. Rather, you would have to remove these edges. For example, conditioning on a collider can open a path.